### ISSUE TITLE
`enforce-switch-style` support for type-switch

### ISSUE THREAD

**Issue by weppos** [mgechev/revive#1628]
**Is your feature request related to a problem? Please describe.**

The enforce-switch-style rule only checks regular switch statements (*ast.SwitchStmt) and ignores type switches (*ast.TypeSwitchStmt). This means a type switch without a default case or with default not in the last position is
  not flagged, even though the same style concerns apply.

```go
// This is flagged (regular switch):
switch x {
  case 1:
}

// This is NOT flagged (type switch):
switch v := x.(type) {
  case int:
}
```

> [!NOTE]
> We found this capability was missing, after we investigated a bug in one of our app where the app was incorrectly dropping some payload, due to a type-switch serialization logic that was not correctly implementing a default fallback for unknown types.

**Describe the solution you'd like**

Extend `enforce-switch-style` to also inspect `*ast.TypeSwitchStmt`. Both switch forms share the same `*ast.BlockStmt` structure, so the existing checks (missing default, default not last, all branches jump) can be applied to type switches with minimal changes.

**Describe alternatives you've considered**

A separate rule dedicated to type switches. However, since the style enforcement logic is identical for both switch forms and they share the same AST body structure, handling them in the same rule avoids duplication and is
  consistent from the user's perspective.

**Additional context**

Both `*ast.SwitchStmt` and `*ast.TypeSwitchStmt` expose a `.Body *ast.BlockStmt` field containing `*ast.CaseClause` entries, so the existing default-position and all-branches-jump checks apply without modification to the inspection logic itself.

Also for context, we use revive via [golangci-lint](https://github.com/golangci/golangci-lint).
